src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCountReader.java

```diff
@@ -0,0 +1,93 @@
+            verbosity = AllelicCountTableColumns.AllelicCountTableVerbosity.FULL;
+        } else if (columns().containsAll(AllelicCountTableColumns.INTERMEDIATE_COLUMN_NAME_ARRAY)) {
+            verbosity = AllelicCountTableColumns.AllelicCountTableVerbosity.INTERMEDIATE;
+        } else if (columns().containsAll(AllelicCountTableColumns.BASIC_COLUMN_NAME_ARRAY)) {
+            verbosity = AllelicCountTableColumns.AllelicCountTableVerbosity.BASIC;
+        } else {
+            final Set<String> missingColumns = Sets.difference(
+                    new HashSet<>(AllelicCountTableColumns.BASIC_COLUMN_NAME_ARRAY), new HashSet<>(columns().names()));
+            throw new UserException.BadInput("Bad header in AllelicCount file. Not all mandatory columns are present." +
+                    " Missing: " + StringUtils.join(missingColumns, ", "));
+        }
+
+    }
+
+    @Override
+    protected AllelicCount createRecord(DataLine dataLine) {
+
+        /* mandatory (basic) fields */
+        final int position = dataLine.getInt(AllelicCountTableColumns.POSITION.name());
+        final SimpleInterval interval = new SimpleInterval(
+                dataLine.get(AllelicCountTableColumns.CONTIG.name()), position, position);
+        final int refReadCount = dataLine.getInt(AllelicCountTableColumns.REF_COUNT.name());
+        final int altReadCount = dataLine.getInt(AllelicCountTableColumns.ALT_COUNT.name());
+
+        /* extra fields */
+        final Nucleotide refNucleotide, altNucleotide;
+        final int readDepth;
+        final double hetLogOdds;
+
+        switch (verbosity) {
+
+            case BASIC:
+                return new AllelicCount(interval, refReadCount, altReadCount);
+
+            case INTERMEDIATE:
+                refNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumns.REF_NUCLEOTIDE.name()).getBytes()[0]);
+                altNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumns.ALT_NUCLEOTIDE.name()).getBytes()[0]);
+                readDepth = dataLine.getInt(AllelicCountTableColumns.READ_DEPTH.name());
+                return new AllelicCount(interval, refReadCount, altReadCount, refNucleotide, altNucleotide, readDepth);
+
+            case FULL:
+                refNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumns.REF_NUCLEOTIDE.name()).getBytes()[0]);
+                altNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumns.ALT_NUCLEOTIDE.name()).getBytes()[0]);
+                readDepth = dataLine.getInt(AllelicCountTableColumns.READ_DEPTH.name());
+                hetLogOdds = dataLine.getDouble(AllelicCountTableColumns.HET_LOG_ODDS.name());
+                return new AllelicCount(interval, refReadCount, altReadCount, refNucleotide, altNucleotide, readDepth,
+                        hetLogOdds);
+
+            default:
+                throw new RuntimeException("The AllelicCount table verbosity level is not properly set.");
```

Comment:
There is two problems with this switch as it stands:

1. There is code that is common to some of the cases that cannot be shared as it is due to the need to chose a constructor in particular.
2. The Default is never going to be executed as is guaranteed that the verbosity is going to be set by the constructor...

Number 2. is not much of a deal by itself although I would recommend to use a more concrete exception (some subclass of GATKException) however I less forgiving with number 1.

I think here you are better of with an if-else-(-if-else-):

```
if (verbosity ==   AllelicCountTableColumns.AllelicCountTableVerbosity.BASIC) {
    return new AllelicCount(interval, refReadCount, altReadCount);
} else {
    final refNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumns.REF_NUCLEOTIDE.name()).getBytes()[0]);
    final altNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumns.ALT_NUCLEOTIDE.name()).getBytes()[0]);
    final readDepth = dataLine.getInt(AllelicCountTableColumns.READ_DEPTH.name());
    if (verbosity == AllelicCountTableColumns.AllelicCountTableVerbosity.INTERMEDIATE) {
         return AllelicCount(interval, refReadCount, altReadCount, refNucleotide, altNucleotide, readDepth); 
    } else { // verbosity == AllelicCountTableColumns.AllelicCountTableVerbosity.FULL;
         final double hetLogOdds = dataLine.getDouble(AllelicCountTableColumns.HET_LOG_ODDS.name());
         return AllelicCount(interval, refReadCount, altReadCount, refNucleotide, altNucleotide, readDepth, hetLogOdds);
    }
}